src/gs-shell.c

```diff
@@ -767,9 +767,6 @@ main_window_closed_cb (GtkWidget *dialog, GdkEvent *event, gpointer user_data)
 	g_application_withdraw_notification (g_application_get_default (),
 					     "install-resources");
 
-	/* When the window is closed, reset the initial mode to overview */
-	priv->mode = GS_SHELL_MODE_OVERVIEW;
-
```

Comment:
Maybe we need to make this conditional on whether the `mode` is `LOADING`. Because the idea for this commit is to always open g-s in the overview mode.

Reply:
This is also handled by GsApplication setting the mode to overview when it activates the window - I think that it's a bad idea to modify the state like this without also updating the other stuff - so this should either use gs_shell_change_mode and do it "properly", or leave it to the GsApplication code - but doing both is really weird to me.

Reply:
That makes sense. I trust you tested reopening the window once it's closed?

Reply:
Yep!